src/backend/distributed/deparser/citus_ruleutils.c

```diff
@@ -1294,6 +1294,20 @@ RoleSpecString(RoleSpec *spec, bool withQuoteIdentifier)
 				   GetUserNameFromId(GetSessionUserId(), false);
 		}
 
+		#if PG_VERSION_NUM >= PG_VERSION_14
```

Comment:
Postgres code and comments makes me think that we should simply treat ROLESPEC_CURRENT_ROLE same as ROLESPEC_CURRENT_USER:
```
Allow CURRENT_ROLE where CURRENT_USER is accepted
In the particular case of GRANTED BY, this is specified in the SQL
standard.  Since in PostgreSQL, CURRENT_ROLE is equivalent to
CURRENT_USER, and CURRENT_USER is already supported here, adding
CURRENT_ROLE is trivial.  The other cases are PostgreSQL extensions,
but for the same reason it also makes sense there.
```

Please read the relevant commit: https://github.com/postgres/postgres/commit/45b9805706fdc726906fc9187c9a7b38c559755c